### ISSUE TITLE
[Cosmos] Bulk Support

### ISSUE THREAD

**Issue by TheovanKraay** [Azure/azure-sdk-for-python#36591]
**Describe the solution you'd like**
Add [bulk support](https://devblogs.microsoft.com/cosmosdb/introducing-bulk-support-in-the-net-sdk/) for Python SDK in Cosmos DB

**Comment by tn-halfspace** [comment]
Yes please, it's a performance bottleneck in many cases for us...

**Comment by kushagraThapar** [comment]
Design choices:
1. Public API consideration: keep it simple.
2. Throughput control: to rate limit bulk operations, and not consume all RUs.
3. Adaptive bactch size: based on workload pattern.
